core/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -420,67 +420,121 @@ class Router(formatOps: FormatOps) {
         }
 
       // Parameter opening for one parameter group. This format works
-      // on a group-by-group basis.
-      case FormatToken(open @ LeftParen(), r, between)
+      // on the WHOLE defnSite (via policies)
+      case FormatToken(open @ (LeftParen() | LeftBracket()), r, between)
           if style.verticalMultilineAtDefinitionSite &&
-            isDefnSite(leftOwner) =>
+            isClassOrDef(leftOwner) =>
+
         val close = matchingParentheses(hash(open))
```

Comment:
Can we move the bulk of this logic outside of Router.scala into a utility file? See for example how `infixSplit` is used. I've noticed that incremental compile times in Router.scala are ~6-7 seconds now and it keeps growing.